Microsoft.Toolkit.Uwp.UI.Controls.Media/ImageCropper/ImageCropper.Events.cs

```diff
@@ -173,6 +173,7 @@ private void ImageCanvas_SizeChanged(object sender, SizeChangedEventArgs e)
 
             UpdateImageLayout();
             UpdateMaskArea();
+            UpdateAspectRatio(true);
```

Comment:
Since `UpdateImageLayout ` and `UpdateMaskArea` will be called in `UpdateAspectRatio ` when KeepAspectRatio == true, `UpdateImageLayout `and `UpdateMaskArea `will actually be called twice here. Maybe we can optimize the logic here

Reply:
I see two options here
1. Add additional checks in `UpdateImageLayout` and `UpdateMaskArea` that prevent it from running unless needed, or
2. Extract only the code needed from `UpdateAspectRatio` into a new method, without the extra method calls.

I'll try for 2)

Reply:
After pulling nested method calls to `UpdateMaskArea` and `UpdateImageLayout` up and out of `UpdateAspectRatio`, then calling them manually / removing the method calls, it seems that updating the image layout is needed before _and_ after updating the AspectRatio.

![image](https://user-images.githubusercontent.com/9384894/185209608-54301c5b-cd54-4faa-a35c-8e0bbfd658c9.png)

There's also some undesirable side effects happening from this refactor that would need to be fixed.
@michael-hawker  optimizing this code path will require more time, what do you advise?